### PR TITLE
Generic layout renderer

### PR DESCRIPTION
**Pull request by danieldelcore**
Adds a new export `LayoutRenderer`. 
This acts as a generic type renderer for cases where the preset layouts don't suit our needs.

The idea is that instead of trying to support every layout and providing override props etc, we just export something that loops over the types and gives consumers a place to substitute their own UI.

This PR also replaces the implementation of the Hybrid layout as a proof of concept.

image

### CODE REVIEW

File: stories/TypeScriptComponent.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ type TypeScriptComponentProps = {
   'quoted-prop': any;
   // @internal
   hideProp: Boolean;
+  nocomment: boolean;
```
Comment (mengX666) [inline stories/TypeScriptComponent.tsx:43]: 👀

Comment (danieldelcore) [reply stories/TypeScriptComponent.tsx:43]: Hah yeah that was intentional to test what happens when a "description" isn't provided :)

File: packages/pretty-proptypes/src/HybridLayout/index.js [hunk 2]
```diff
@@ -4,93 +4,151 @@
...
+          renderType={({
+            typeValue,
+            defaultValue,
+            description,
+            required,
+            name,
+            type,
+            components: Comp
+          }) => (
+            <table
```
Comment (AndrewOCC) [inline packages/pretty-proptypes/src/HybridLayout/index.js:53]: Nit: Perhaps we should split this component out into a separate file like PropEntry was before?

File: packages/pretty-proptypes/src/renderPropType.js [hunk 3]
```diff
@@ -81,23 +79,20 @@ const renderPropType = (
...
+      key={name}
+      components={components}
+      name={name}
+      description={description}
+      required={!propType.optional}
+      type={getKind(propType.value)}
+      defaultValue={propType.default && convert(propType.default)}
+      shouldCollapse={shouldCollapseProps}
+      typeValue={propType.value}
+    />
```
Comment (AndrewOCC) [inline packages/pretty-proptypes/src/renderPropType.js:95]: Nice and neat 😍

**Review by AndrewOCC — COMMENTED** [review]
Looks great - love that there's a neater way for consumers to customise the prop output. Left a few small comments.

**Review by AndrewOCC — APPROVED** [review]

**Pull request merged**